# Project X -- [Prisha ] [Joshi]

**TA Help:** John Smith, Alice Jones

- Help with figuring out how to write a function.
    
**Collaboration:** Friend1, Friend2
    
- Helped figuring out how to load the dataset.
- Helped debug error with my plot.

## Question 1

In [1]:
# code here
library(data.table)
df <- fread("/anvil/projects/tdm/data/death_records/DeathRecords.csv")
makeatable <- function(x) {prop.table(table(x, useNA="always"))}
makeatable(df$Sex)
makeatable(df$MaritalStatus)



x
        F         M      <NA> 
0.4939664 0.5060336 0.0000000 

x
          D           M           S           U           W        <NA> 
0.152388043 0.372463819 0.126575962 0.007112043 0.341460133 0.000000000 

The table for death records is as shown. 49.39% females and 50.61% males are in the death records. 15.24% OF PEOPLE WERE DIVORCED, 37.25% WERE MARRIED, 12.65 WERE SINGLE , 7% WERE UNDER THE TITLE "U" (possibly unmarried ), AND 34.14% WERE WIDOWED 

## Question 2

In [3]:
# code here
library(data.table)
df1 <- fread("/anvil/projects/tdm/data/olympics/athlete_events.csv")
teenagecount <- function(x) {length(x[(x >= 13) & (x <= 19) & (!is.na(x))])}
teenagecount(df$Age)
teenagecount(df1$Age)

[1] 12643

[1] 32250

12643 teenagers of the given age are present in death records data. Those many teenagers withing the age 13 and 19 died that year. 32250 teenagers were athletes in the olympics data. 

## Question 3

In [6]:
library(data.table)
myDF <- fread("/anvil/projects/tdm/data/election/itcont1980.txt", quote="")
names(myDF) <- c("CMTE_ID", "AMNDT_IND", "RPT_TP", "TRANSACTION_PGI", "IMAGE_NUM", "TRANSACTION_TP", "ENTITY_TP", "NAME", "CITY", "STATE", "ZIP_CODE", "EMPLOYER", "OCCUPATION", "TRANSACTION_DT", "TRANSACTION_AMT", "OTHER_ID", "TRAN_ID", "FILE_NUM", "MEMO_CD", "MEMO_TEXT", "SUB_ID")
longesttest <- function(x) {x[which.max(nchar(x))]}
df2 <- fread("/anvil/projects/tdm/data/beer/reviews_sample.csv")
longesttest(df2$text)
longesttest(myDF$NAME)


[1] "\302\240\302\240 32 fl oz brown glass Alesmith grumbler filled earlier today at The Growler Room in Austin, Texas, and stored in my fridge until consumption. Served into a Spaten stangekrug. Reviewed at low altitude live here in Austin on 08/31/13. Cost was $ $21 USD. Expectations are sky high given both its current ratings and the description \342\200\223 which makes it sound tasty. This is my first Texan sour. Served frigid cold and allowed to warm following the initial pour. Side-poured with extra vigor as slight carbonation issues are anticipated given that it\342\200\231s a growler pour. A: No bubble show forms as I pour. Pours a one finger wide head of gorgeous khaki colour. Great creaminess \342\200\223 reminiscent actually of an Irish Dry Stout\342\200\231s head. Good thickness. Supple. Has an appealing soft complexion. Okay lacing; some legs cling to the sides of the glass when I tilt it, but they quickly fall. Retention is pretty good \342\200\223 about 3-4 minutes; not too shabby for a growler pour. Body colour is a nontransparent murky caramel-brown. Some light glowing translucency near the bottom of the glass; it\342\200\231s certainly not opaque. No yeast particles are visible. It\342\200\231s not incredible to look at, but there\342\200\231s nothing overtly wrong with it. Looking forward to trying it. Sm: Tart acetic cherries, cherry pie crust, and accompanying sourness. It\342\200\231s got more of a Flanders Red feel than anything else. Very light funkiness. I\342\200\231m very impressed with the yeast-derived sourness and I\342\200\231m rather curious what strain was used; this doesn\342\200\231t have the sterile clinical feel of a sour brewed purely with lactobacillus (at least based on the aroma) \342\200\223 there\342\200\231s more complexity here, and the combination of well-integrated sourness with a subtle funkiness and complex tart character approximates lambic yeast. The yeast alone elevates the aroma beyond that of most wild ales. Unlike many subpar \342\200\234wild\342\200\235 ales, this one actually smells like it\342\200\231s still alive \342\200\223 and I don\342\200\231t find myself doubting that wild yeast and bacteria were used in fermentation. It stops a bit short of evoking true spontaneous open fermentation, but I\342\200\231m nitpicking. Very buried hints of vanilla and maple; they\342\200\231re by no means overt. Layered caramel and biscuit malts. As it warms, the otherwise reticent oak notes begin to creep out; I don\342\200\231t know that I\342\200\231d say they\342\200\231re evocative of a barrel per se. The cherry notes are dominant, a sound choice. No alcohol is detectable, nor is any significant hop character. A damn fine aroma for an American sour. Mild in strength, with well-executed subtlety. I\342\200\231m impressed. I just hope the taste lives up to the smell. T: Tart cherries, acetic cherry pie crust notes (though not as prominent as I\342\200\231d like), and a pleasant core sourness throughout are the first things I notice. It\342\200\231s got a clean simple open comprised mainly of biscuit malts with a dash of caramalt in there too, a malty combination that continues on through the second act to form the body of the beer. Luckily, the malts don\342\200\231t intrude and the beer\342\200\231s sourness is allowed to take centre stage. Speaking of the sourness, it\342\200\231s by no means puckerworthy. It\342\200\231s somewhat tame yet still satisfying. This would be a good introduction to (American) sours for a beginner. Acidic. I have to retract what I said about the yeast above; it by no means approximates lambic yeast in the actual taste, and does suffer from the same sterile, clean, clinical feel of most American wild ales/sours. Unfortunate. Oak is present as a guiding undertone. Luckily, those vanilla and maple notes from the aroma are near imperceptible. No hop character or alcohol comes through. It\342\200\231s got a good balance to it, and it\342\200\231s nearest a

[1] "REPUBLICAN NATIONAL COMMITTEE - CONTRIBUTIONS (AKA REPUBLICAN NATIONAL FINANCE COMMITTEE"

the longest review is as shown above. The longest name in the elections data was the 'republican national committee - contributions (aka republican national finance committee'

## Question 4

In [8]:
mostpopulardate <- function(x) {x[which.max(nchar(x))]}
mostpopulardate(df2$date)
mostpopulardate(myDF$TRANSACTION_DT)

[1] "2017-03-17"

[1] 10031979

The most popular date in the beer reviews dataset is 17th march 2017. There were most of the reviews written on that date. The most popular date for the elections data is 10-03-1979. The most transactions were done on that date. 

## Question 5

In [1]:
library(data.table)
df3 <- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")
myaveragedelay <- function(x) {mean(df3$DepDelay[df3$Origin == x],na.rm =TRUE)}
myaveragedelay("IND")
myaveragedelay("JFK")

[1] 5.969772

[1] 11.85727

The average delay for departure delay of flight from Indianapolis "IND" is 5.968 hours. The average delay for the departure delay of flight from JFK is 11.8573 hours. 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.